crates/runtime/src/dataconnector.rs

```diff
@@ -278,3 +305,114 @@ pub async fn get_data(
 
     Ok((table_provider.schema(), batches))
 }
+
+pub trait ListingTableConnector: DataConnector {
```

Comment:
Can't we just have a `pub Struct ListingTableConnector` rather than trait that has a method `get_object_store_url` which based on the object_store_type to decide how to construct the url?

Given the most abstraction is done in object_store level, I don't think we need to have another level of abstraction using the trait here.

Reply:
unless there are some really big advantage this trait can provide.

Reply:
I think it's more readable this way and more inline with how we implement other data connectors